File: C#/TddApi/NavigationApp/s11.refactor_test_with_mock/Navigator.cs
using System;

namespace NavigationApp.s11.refactor_test_with_mock
{
    public class Navigator
    {
        DistanceProvider distProvider;
        Location startingPoint;

        public Navigator(Location loc, DistanceProvider distService)
        {
            distProvider = distService;
            startingPoint = loc;
        }

        public Distance getDistanceFromDestination()
        {
            return distProvider.getDistance(startingPoint, startingPoint);
        }

        public void setDestination(Location dest)
        {
                
        }
    }
}

File: Python/s12_re_add_second_test/navigator.py
class Navigator:
    def __init__(self, starting_point, dist_provider):
        self.dist_provider = dist_provider
        self.starting_point = starting_point

    def get_distance_from_destination(self):
        return self.dist_provider.get_distance(self.starting_point, self.starting_point)

    def setDestination(self, dest):
        pass
